### PR TITLE
Added continuous fragment mode

### PR DESCRIPTION
**Pull request by pablogs9**
Before merge:

- [x] Remove CI branch of XRCE Client when https://github.com/eProsima/Micro-XRCE-DDS-Client/pull/177 merged
- [ ] Review in detail @jamoralp

### CODE REVIEW

**Review by jamoralp — APPROVED** [review]

File: rmw_microxrcedds_c/test/test_publisher.cpp [hunk 1]
```diff
@@ -271,3 +273,86 @@ TEST_F(TestPublisher, memory_poll_shared_topic) {
     publishers.clear();
   }
 }
+
+/*
+   Testing continous fragment mode
+ */
+
+TEST_F(TestPublisher, continous_fragment_mode) {
```
Comment (pablogs9) [inline rmw_microxrcedds_c/test/test_publisher.cpp:281]: This is not using `rmw_uros_set_continous_serialization_callbacks`

?????

**Comment by pablogs9** [comment]
This PR should target `foxy`

File: rmw_microxrcedds_c/include/rmw_uros/options.h [hunk 2]
```diff
@@ -89,6 +91,18 @@ rmw_ret_t rmw_uros_options_set_client_key(uint32_t client_key, rmw_init_options_
  */
 rmw_ret_t rmw_uros_check_agent_status(int timeout_ms);
 
+/**
+ * \brief Sets the callback functions for continous serialization
+ *
+ * \param[in] TODO
```
Comment (jamoralp) [inline rmw_microxrcedds_c/include/rmw_uros/options.h:97]: docs

Comment (pablogs9) [reply rmw_microxrcedds_c/include/rmw_uros/options.h]: Done

**Comment by pablogs9** [comment]
I have linked callback with a certain publisher in order to manage it independently for each entity

**Review by jamoralp — APPROVED** [review]

**Pull request merged**